lib/taskana-core/src/test/java/acceptance/task/UpdateManualPriorityWithSpiAccTest.java

```diff
@@ -0,0 +1,143 @@
+
+@TaskanaIntegrationTest
+@WithServiceProvider(
+    serviceProviderInterface = PriorityServiceProvider.class,
+    serviceProviders = TestStaticValuePriorityServiceProvider.class)
+class UpdateManualPriorityWithSpiAccTest {
+
+  @TaskanaInject TaskanaEngine taskanaEngine;
+  @TaskanaInject TaskService taskService;
+  @TaskanaInject ClassificationService classificationService;
+  @TaskanaInject WorkbasketService workbasketService;
+
+  ClassificationSummary defaultClassificationSummary;
+  WorkbasketSummary defaultWorkbasketSummary;
+  ObjectReference defaultObjectReference;
+  TaskUpdatePriorityWorker worker;
+
+  @WithAccessId(user = "businessadmin")
+  @BeforeAll
+  void setup() throws Exception {
+    worker = new TaskUpdatePriorityWorker(taskanaEngine);
+    defaultClassificationSummary =
+        defaultTestClassification().priority(9).buildAndStoreAsSummary(classificationService);
+    defaultWorkbasketSummary = defaultTestWorkbasket().buildAndStoreAsSummary(workbasketService);
+
+    WorkbasketAccessItemBuilder.newWorkbasketAccessItem()
+        .workbasketId(defaultWorkbasketSummary.getId())
+        .accessId("user-1-1")
+        .permission(WorkbasketPermission.OPEN)
+        .permission(WorkbasketPermission.READ)
+        .permission(WorkbasketPermission.APPEND)
+        .buildAndStore(workbasketService);
+    defaultObjectReference = defaultTestObjectReference().build();
+  }
+
+  @WithAccessId(user = "user-1-1")
+  @Test
+  void should_setPriorityToManualAndNotUpdateAccordingToSpi_When_ManualPriorityPositive()
+      throws Exception {
+    Task task = createDefaultTask().manualPriority(123).buildAndStore(taskService);
+    worker.executeBatch(List.of(task.getId()));
+    Task result = taskService.getTask(task.getId());
+    assertThat(result.getPriority()).isEqualTo(result.getManualPriority()).isEqualTo(123);
+  }
+
+  @WithAccessId(user = "user-1-1")
+  @Test
+  void should_setPriorityUsingSpi_When_ManualPriorityNegative() throws Exception {
+    Task task = createDefaultTask().manualPriority(-1).buildAndStore(taskService);
+    Task result = taskService.getTask(task.getId());
+    assertThat(result.getPriority()).isEqualTo(5);
+  }
+
+  @WithAccessId(user = "user-1-1")
+  @Test
+  void should_UpdatePriorityAccordingToSpi_When_UpdatedTaskHasNegativeManualPriority()
+      throws Exception {
+    Task task = createDefaultTask().manualPriority(123).buildAndStore(taskService);
+    task.setManualPriority(-3);
+    taskService.updateTask(task);
+    Task result = taskService.getTask(task.getId());
+    assertThat(result.getPriority()).isEqualTo(5);
+  }
+
+  @WithAccessId(user = "user-1-1")
+  @Test
+  void should_UpdateManualPriorityAnd_When_UpdatedTaskHasZeroManualPriority() throws Exception {
+    Task task = createDefaultTask().manualPriority(123).buildAndStore(taskService);
+    task.setManualPriority(0);
+    taskService.updateTask(task);
+    Task result = taskService.getTask(task.getId());
```

Comment:
same as the previous test case. The additional fetching of the task is not necessary.